### ISSUE TITLE
Suspicious connection to a js behind-the-scene

### ISSUE THREAD

**Issue by uBlock-user** [ChromeDevTools/devtools-frontend#33]
Hi, I'm running Chromium 63, lately upon visiting any webpage and opening devtools via rightclick and clicking on inspect a behind-the-scenes network connection request is fired by Chrome trying to load a suspicious JS. I will attach a screenshot below.

![](https://i.gyazo.com/53667c56920f415c273effa0e1a28f9e.png)

This is the JS that tries to load - https://chrome-devtools-frontend.appspot.com/serve_file/@6263ab04beb58e7ed19b027f211d080aa954dc74/product_registry_impl/product_registry_impl_module.js

What is it, what does it do and why did it start happening since Chromium 63 ?

**Comment by uBlock-user** [comment]
11 days and not a single response!! Either I'm being ignored purposefully or the team has no interest in answering the issue tracker as expected.

**Comment by paulirish** [comment]
I believe product_registry_impl_module.js requests started in chromium 60, not 63. 
The feature writeup is here: https://developers.google.com/web/updates/2017/05/devtools-release-notes#badges

 DevTools loads some things off of a CDN in order to keep the binary size of chromium down. The product registry data, device images (if you turn on Show Device Frame while emulating) and the Auditing engine (audits panel, start) will all load off of our appspot domain.

You're welcome to block these requests if you want, but they're not harmful. We own the domain and the data there.

**Comment by paulirish** [comment]
@uBlock-user take a moment to click through to that feature writeup. It's actually quite relevant to you as a ublock user. :) 
It won't work well in concert with ublock, but it would be an interesting feature to enable when you have ublock disabled.

**Comment by uBlock-user** [comment]
> I believe product_registry_impl_module.js requests started in chromium 60, not 63.

I didn't see this in 60 or 61 or 62, it only started happening on my end since 63.

@paulirish I opened the JS and I see names of lots and lots of Adservers/companies, can you justify that ?

**Comment by uBlock-user** [comment]
For example - "PropellerADs media Ltd"  

Does that JS track me and send the tracked data to those adserving companies ? OR 

is it used to serve ads in chrome via behind-the-scenes in future or in near future ?

It will stay blocked on my end until my questions are answered.

**Comment by uBlock-user** [comment]
Also why did you remove the screenshot ? It was there when I posted it.

**Comment by paulirish** [comment]
@uBlock-user  FWIW i use uBlock too. IMO, DevTools is a neutral player when it comes to ads and tracking, etc.

> Also why did you remove the screenshot ? It was there when I posted it.

I still see a screenshot on this ticket.  Seems OK.


> Does that JS track me and send the tracked data to those adserving companies ? 

nope. definitely not.

> Will it be used to serve ads in chrome via behind-the-scenes in future or in near future ?

no. you might have heard news about a chrome ad blocker. chrome is far more excited to do that than to serve ads built into the browser.

> I see names of lots and lots of Adservers/companies, can you justify that ?

Here's a video where I describe the feature you're looking at: https://youtu.be/PjjlwAvV8Jg?t=1619  26:59 to 28:38

Basically we added a feature to devtools to help developers decrypt all the third-party network requests going on and figure out who each domain is. This is so developers who see their site loading slowly can now point the finger at the correct 3rd party.

Does that make sense?

**Comment by paulirish** [comment]
And lastly, a devtools user has to enable this 3rd party attribution feature to see it in action. And since you'll keep uBlock on, I suspect nearly all the requests will be blocked anyway. So the feature won't really provide much benefit for you. (which is fine)

Here's the original source of that JS: https://github.com/ChromeDevTools/devtools-frontend/tree/master/front_end/product_registry_impl

Since you won't be using the feature, you're very welcome to keep blocking the script. The 3rd party attribution feature will remain disabled, and hopefully everything else should work as expected.

I just wanted to clarify that there's nothing shady going on, and quite the opposite, the feature was introduced to make things less shady. :)

**Comment by uBlock-user** [comment]
> Basically we added a feature to devtools to help developers decrypt all the third-party network requests going on and figure out who each domain is. This is so developers who see their site loading slowly can now point the finger at the correct 3rd party.

That statement justifies it properly as I have seen the waterfall in devtools and also seen how it shows the name of the ad-serving company if I hover my mouse on the coloured waterfall(for a lack of a better word).

**Comment by paulirish** [comment]
Yup! that's correct.

**Comment by uBlock-user** [comment]
Thanks for the clearing up the mis-understanding.
